pkg/sql/create_view.go

```diff
@@ -116,6 +116,22 @@ func (n *createViewNode) startExec(params runParams) error {
 		case !sqlerrors.IsRelationAlreadyExistsError(err):
 			return err
 		case n.ifNotExists:
+			// Get the descriptor to check if its in a dropping state
+			id, err := catalogkv.GetDescriptorID(params.ctx, params.p.txn, params.ExecCfg().Codec, tKey)
+			if err != nil {
+				return err
+			}
+			desc, err := params.p.Descriptors().GetMutableTableVersionByID(params.ctx, id, params.p.txn)
+			if err != nil {
+				return err
+			}
+
+			if desc.Dropped() {
+				return pgerror.Newf(pgcode.ObjectNotInPrerequisiteState,
+					"view %q is being dropped, try again later",
```

Comment:
I believe it's possible that `desc` isn't a view, for example:
```
CREATE TABLE foo (a INT);
BEGIN;
  DROP TABLE IF EXISTS foo;
  CREATE VIEW IF NOT EXISTS foo AS SELECT 1;
COMMIT;
```

Reply:
Good find, yes I need to check IsView on the descriptor.